serializable/helpers.py

```diff
@@ -203,9 +203,9 @@ def deserialize(cls, o: Any) -> datetime:
                 # Remove any leading hyphen
                 v = v[1:]
 
-            # Ensure any milliseconds are 6 digits
-            # Background: py<3.11 supports six or less digits for milliseconds
-            v = re_sub(r'\.(\d{1,6})', lambda m: f'.{int(m.group()[1:]):06}', v)
+            # Ensure either 0 or exactly 6 decimal places for seconds
+            # Background: py<3.11 supports either 6 or 0 digits for milliseconds
+            v = re_sub(r'(\.\d{1,6})\d*', lambda m: f'{(float(m.group(0))):.6f}'[1:], v)
```

Comment:
I see, you are going with group `0`, instead of group `1`, and parse it all to `float`, so rounding is applied.
A very good idea.

In this case, we could even improve it, I guess:
```py
v = re_sub(r'\.\d+', lambda m: f'{(float(m.group(0))):.6f}'[1:], v)
```

Reply:
done via https://github.com/madpah/serializable/pull/138/commits/88c90b009fdaa2b209ea4bd7d05d85cc7a281f08